File: src/main/java/com/aprilboiz/flightbookingsite/service/AirportService.java
package com.aprilboiz.flightbookingsite.service;

import com.aprilboiz.flightbookingsite.entity.Airport;
import com.aprilboiz.flightbookingsite.repository.AirportRepository;
import org.springframework.stereotype.Service;

@Service
public class AirportService {
    private final AirportRepository airportRepository;

    public AirportService(AirportRepository airportRepository) {
        this.airportRepository = airportRepository;
    }

    public Airport getAirportByName(String airportName) {
        return airportRepository.findAirportByName(airportName).orElse(null);
    }
}

File: internal/service/airport_service.go
package service

import (
	"github.com/aprilboiz/flight-management/internal/models"
	"github.com/aprilboiz/flight-management/internal/repository"
)

type airportService struct {
	airportRepo repository.AirportRepository
}

func NewAirportService(airportRepo repository.AirportRepository) AirportService {
	return &airportService{airportRepo: airportRepo}
}

func (a airportService) GetAllAirports() ([]*models.Airport, error) {
	airports, err := a.airportRepo.GetAll()
	if err != nil {
		return nil, err
	}
	return airports, nil
}

func (a airportService) GetAirportByCode(code string) (*models.Airport, error) {
	airport, err := a.airportRepo.GetByCode(code)
	if err != nil {
		return nil, err
	}
	return airport, nil
}

func (a airportService) GetAirportsByCodes(codes []string) (map[string]*models.Airport, error) {
	airports, err := a.airportRepo.GetByCodes(codes)
	if err != nil {
		return nil, err
	}
	return airports, nil
}
